packages/story-editor/src/app/rightClickMenu/hooks/useCopyPasteActions.js

```diff
@@ -0,0 +1,238 @@
+    const id = selectedElement?.id;
+
+    if (!id || selectedElement?.type !== copiedElement.type) {
+      return;
+    }
+
+    // Delete old animation if one exists
+    const oldAnimationToDelete = selectedElementAnimations.length
+      ? { ...selectedElementAnimations[0], delete: true }
+      : undefined;
+
+    // Create new animations
+    const newAnimations = copiedElement.animations.map((animation) => ({
+      ...animation,
+      id: uuidv4(),
+      targets: [selectedElement.id],
+    }));
+
+    addAnimations({ animations: newAnimations });
+
+    // Text elements need the text styles extracted from content before
+    // applying to the other text
+    if (copiedElement.type === 'text' && copiedElement.styles.content) {
+      const { textStyles } = getTextPresets(
+        [copiedElement.styles],
+        {
+          textStyles: [],
+          colors: [],
+        },
+        PRESET_TYPES.STYLE
+      );
+      const { colors } = getTextPresets(
+        [copiedElement.styles],
+        {
+          textStyles: [],
+          colors: [],
+        },
+        PRESET_TYPES.Color
+      );
+      const { content, ...copiedElementStyles } = copiedElement.styles;
+      const updatedElementStyles = {
+        ...copiedElementStyles,
+        ...textStyles[0],
+        ...colors[0].color,
+        animation: oldAnimationToDelete,
+        border: copiedElementStyles.border || null,
+      };
+      handleApplyStyle(updatedElementStyles);
+    } else {
+      // Add styles and animations to element
+      updateElementsById({
+        elementIds: [selectedElement.id],
+        properties: (currentProperties) =>
+          updateProperties(
+            currentProperties,
+            {
+              ...copiedElement.styles,
+              animation: oldAnimationToDelete,
+            },
+            /* commitValues */ true
+          ),
+      });
+    }
+
+    showSnackbar({
+      actionLabel: __('Undo', 'web-stories'),
+      dismissible: false,
+      message: __('Pasted style.', 'web-stories'),
+      // don't pass a stale reference for undo
+      // need history updates to run so `undo` works correctly.
+      onAction: () => {
+        undoRef.current();
+
+        trackEvent('context_menu_action', {
+          name: 'undo_paste_styles',
+          element: selectedElement?.type,
+          isBackground: selectedElement?.isBackground,
+        });
+      },
+      actionHelpText: UNDO_HELP_TEXT,
+    });
+
+    trackEvent('context_menu_action', {
+      name: 'paste_styles',
+      element: selectedElement?.type,
+      isBackground: selectedElement?.isBackground,
+    });
+  }, [
+    addAnimations,
+    copiedElement,
+    handleApplyStyle,
+    selectedElement,
+    selectedElementAnimations,
+    showSnackbar,
+    updateElementsById,
+  ]);
+
+  return {
+    copiedElementType: copiedElement?.type,
+    onCopyStyles: handleCopyStyles,
+    onPasteStyles: handlePasteStyles,
```

Comment:
I hate critiquing naming conventions, but just gonna drop a small not on my mental model of `on` vs `handle`

I'd be fine with these just being `handle...` vs `on...`

My reasoning for this is that `on..` is normally associated as an event that you can tap into, `handle...` is normally associated with the action that taps into that event. 

This hook is creating actions that handle how the copy or paste function, not exposing when copy and paste fire